packages/recommend-react/src/__tests__/useFrequentlyBoughtTogether.test.ts

```diff
@@ -37,9 +43,98 @@ describe('useFrequentlyBoughtTogether', () => {
           facetFilters: ['test'],
         },
         transformItems: (items) => items,
-      });
+      })
+    );
 
-      expect(recommendations).toEqual([hit]);
+    await waitFor(() => {
+      expect(result.current.recommendations).toEqual([hit]);
     });
   });
+  test('gets fbt products from initialState', async () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    const { result } = renderHook(
+      () =>
+        useFrequentlyBoughtTogether({
+          indexName: 'test',
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: React.StrictMode,
+      }
+    );
+
+    await waitFor(() => {
+      expect(result.current.recommendations).toEqual(
+        initialState.recommendations
+      );
+    });
+  });
+
+  test('no network calls with initialState', () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    renderHook(
+      () =>
+        useFrequentlyBoughtTogether({
+          indexName: 'test',
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: React.StrictMode,
+      }
+    );
+
+    expect(recommendClient.getFrequentlyBoughtTogether).toHaveBeenCalledTimes(
+      0
+    );
+  });
+
+  test('trigger network call when props change', async () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    const { rerender } = renderHook(
+      ({ indexName }) =>
+        useFrequentlyBoughtTogether({
+          indexName,
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: React.StrictMode,
+        initialProps: { indexName: 'test' },
+      }
+    );
+    expect(recommendClient.getFrequentlyBoughtTogether).toHaveBeenCalledTimes(
+      0
+    );
+
+    await forceDelay(1000);
```

Comment:
Why do we need to have a delay? Leaving context either as a code comment or in this PR would help reviewers understand this.

Reply:
✅

Reply:
I don't understand why this is needed.

Should we rather call `rerender()` and then `waitFor()` the `expect`? (Or perhaps we need to wrap this `rerender()` in `act()`.)

Reply:
✅